lib/src/subwasm.rs

```diff
@@ -58,77 +59,14 @@ impl Subwasm {
 	// 	Ok(())
 	// }
 
-	pub fn display_module(&self, filter: String) {
-		let metadata = self.testbed.runtime_metadata_prefixed();
-		let wrapper = MetadataWrapper(&metadata.1);
-		wrapper.display_single_module(&filter);
-	}
-
-	pub fn display_modules_list(&self) {
+	pub fn write_metadata<O: Write>(
+		&self,
+		fmt: metadata_wrapper::OutputFormat,
+		filter: Option<String>,
+		out: &mut O,
+	) -> Result<(), Box<dyn std::error::Error>> {
 		let metadata = self.testbed.runtime_metadata_prefixed();
 		let wrapper = MetadataWrapper(&metadata.1);
-		wrapper.display_modules_list();
-	}
-
-	/// Display the metadata as json
-	pub fn display_metadata_json(&self) {
-		let pallet_filter: Option<String> = None;
-		let metadata = self.testbed.metadata();
-
-		let serialized = if let Some(ref pallet) = pallet_filter {
-			match metadata {
-				RuntimeMetadata::V12(v12) => {
-					let modules = convert(&v12.modules);
-
-					let pallet_metadata = modules
-						.iter()
-						.find(|module| module.name == DecodeDifferent::Decoded(pallet.into()))
-						.expect("pallet not found in metadata");
-					serde_json::to_string_pretty(&pallet_metadata)
-				}
-				// RuntimeMetadata::V13(v13) => {
-				// 	let pallet = v13
-				// 		.modules
-				// 		.iter()
-				// 		.find(|m| &m.name == pallet)
-				// 		.ok_or_else(|| eyre::eyre!("pallet not found in metadata"))?;
-				// 	serde_json::to_string_pretty(&pallet)?
-				// }
-				_ => panic!("Unsupported metadata version"),
-			}
-		} else {
-			serde_json::to_string_pretty(&metadata)
-		};
-
-		// The following fails if piped to another command that truncates the output.
-		// Typical use case here is: subwasm meta | head
-		// The failure is due to https://github.com/rust-lang/rust/issues/46016
-		// TODO: Once the above is fixed, we can remove the dependency on calm_io
-		// println!("{}", serialized);
```

Comment:
Did you test that part ?
I see the [issue](https://github.com/rust-lang/rust/issues/46016) is still open.

Reply:
The root cause of this issue is that we can't ignore the broken pipe error in a println! call. Since all printing is now done through writeln macro calls I just improved error propagation and added a top-level check to ignore the broken pipe error. I consequently removed the calm_io dependency.

While doing so I saw you already started using color_eyre, so I properly installed it in the beginning of main and also replaced a lot of unwraps and panics with better error propagation. 

You will be happy seeing the next error. That eyre crate is really nice, I didn't used it before.

Reply:
If you do `RUST_LOG=subwasm=debug` you can see a debug line when the broken pipe error is ignored.